tests/test_api.py

```diff
@@ -357,28 +383,26 @@ def catch(*args, **kwargs):
             self.assertEqual(iface, "org.freedesktop.Test.Main")
 
             changed_props.append(changed)
-            ml.quit()
+
+            if len(changed_props) == len(changes):
+                ml.quit()
 
         match = self.dbus_con.add_signal_receiver(
             catch, interface_keyword="interface", path_keyword="path", member_keyword="member"
         )
 
         # change property using mock helper
-        self.dbus_mock.UpdateProperties(
-            "org.freedesktop.Test.Main",
-            {
-                "version": 5,
-                "connected": False,
-            },
-        )
+        for change in changes:
+            self.dbus_mock.UpdateProperties("org.freedesktop.Test.Main", change)
+            self.assertTrue(change.items() <= self.dbus_props.GetAll("org.freedesktop.Test.Main").items())
 
         GLib.timeout_add(3000, ml.quit)
         ml.run()
 
         match.remove()
 
-        self.assertEqual(self.dbus_props.GetAll("org.freedesktop.Test.Main"), {"version": 5, "connected": False})
-        self.assertEqual(changed_props, [{"version": 5, "connected": False}])
+        for i, _ in enumerate(changes):
+            self.assertEqual(changed_props[i], changes[i])
```

Comment:
Could be a bit more elegant with `for i, change in ..` and comparing against `change`, but that's total bikeshedding.

Reply:
Lol, that's exactly what [pylint complains about](https://github.com/martinpitt/python-dbusmock/actions/runs/15463734765/job/43530844659?pr=239#step:4:848). So can you please do this still?